client/lib/site/utils.js

```diff
@@ -64,6 +64,10 @@ export default {
 		}
 	},
 
+	hasCustomDomain( site ) {
```

Comment:
I'm afraid we can't get all domains for a site directly from `site` property. Ideally we should get the domains making a request to its endpoint.